samples/original-manual-adjusted.rb

```diff
@@ -30,7 +30,7 @@ def dewikify_hi(str, terms, intro = false)
       str = str
         .gsub(CODE_RE) { |x| code << x; "CODE#[#{code.length-1}]" }
         .gsub(/#{Regexp.quote(terms)}/i, '@\0@')
-        .gsub(/CODE#\[(\d+)\]/) { code[$1.to_i] }
+        .gsub(/CODE#\[(\d+)\]/) { code[Regexp.last_match(1).to_i] }
```

Comment:
While not a huge fan of $1, this seems worse to me. If we want to revise to it, I'd prefer to explicitly call match and use the return vs Regexp.last_match. @PragTobi what do you think?